src/System/OS.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Dock\System;
+
+class OS
+{
+    const MAC = 1;
+    const LINUX = 2;
+    const WINDOWS = 3;
+
+    /**
+     * @return int Current Operating System
+     */
+    public function get()
```

Comment:
I'm strongly against `isLinux` and `isMac` methods. Plus the method is also used here, and once we abstract away the `createNewShell()` it'll also be used from wherever that was abstracted to.